File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/sed/medlemskap/impl/MedlemskapA008.java
package no.nav.melosys.eessi.models.sed.medlemskap.impl;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import no.nav.melosys.eessi.models.sed.medlemskap.Medlemskap;
import no.nav.melosys.eessi.models.sed.nav.EndringA008;
import no.nav.melosys.eessi.models.sed.nav.MedlemskapA008Bruker;

@EqualsAndHashCode(callSuper = false)
@JsonIgnoreProperties(ignoreUnknown = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
public class MedlemskapA008 implements Medlemskap {

    private EndringA008 endring;

    private MedlemskapA008Bruker bruker;
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/sed/medlemskap/impl/MedlemskapA008.kt
package no.nav.melosys.eessi.models.sed.medlemskap.impl

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import no.nav.melosys.eessi.models.sed.medlemskap.Medlemskap
import no.nav.melosys.eessi.models.sed.nav.EndringA008
import no.nav.melosys.eessi.models.sed.nav.MedlemskapA008Bruker

@JsonIgnoreProperties(ignoreUnknown = true)
data class MedlemskapA008(
    var endring: EndringA008? = null,
    var bruker: MedlemskapA008Bruker? = null,
    var formaal: String? = null // CDM 4.4: "endringsmelding" | "arbeid_flere_land"
) : Medlemskap
